### PR TITLE
[INLONG-9009][Sort] Add HBase source and sink connector on flink 1.15

### LINKED ISSUE #9009: [Feature][Sort] Add HBase connector on flink 1.15
### Description

Parent issue: https://github.com/apache/inlong/issues/8336
Add HBase connector on flink 1.15

### Use case

Add HBase connector on flink 1.15

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fancycoderzf**
### Prepare a Pull Request

- Fixes #9009 

### Motivation

*Add HBase source and sink connector on flink 1.15*

### Modifications

*Add HBase source and sink connector on flink 1.15*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

#####################################################################################################
I have some trouble to write the end-to-end test for HBase. So I just run some tests in flink(v1.15) bin/sql-client.sh. I will consider adding end-to-end testing.
Sink:
image
Source:
image
image

### CODE REVIEW

**Comment by EMsnap** [comment]
Please add some tests on this connector thanks

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hbase/pom.xml [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                            </filters>
+                            <relocations>
+                                <!-- Force relocation of all HBase dependencies. -->
+                                <relocation>
+                                    <pattern>org.apache.inlong.sort.base</pattern>
+                                    <shadedPattern>org.apache.inlong.sort.hbase.shaded.org.apache.inlong.sort.base</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>org.apache.zookeeper</pattern>
+                                    <shadedPattern>org.apache.flink.hbase.shaded.org.apache.zookeeper</shadedPattern>
```
Comment (vernedeng) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hbase/pom.xml:164]: Please relocate these patterns under **org.apache.inlong.sort.xx** dictionary

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hbase/pom.xml [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-common</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.inlong</groupId>
+            <artifactId>sort-format-common</artifactId>
+            <version>1.10.0-SNAPSHOT</version>
```
Comment (vernedeng) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hbase/pom.xml:72]: Use ${project.version}

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hbase/pom.xml (hunk 1 above)
Comment (fancycoderzf) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hbase/pom.xml]: done

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hbase/src/main/java/org/apache/inlong/sort/hbase/HBase2DynamicTableFactory.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+    @Override
+    public DynamicTableSource createDynamicTableSource(Context context) {
+        FactoryUtil.TableFactoryHelper helper = createTableFactoryHelper(this, context);
+        helper.validateExcept(PROPERTIES_PREFIX);
+
+        final ReadableConfig tableOptions = helper.getOptions();
+
+        DataType tableSchema = context.getCatalogTable().getResolvedSchema().toPhysicalRowDataType();
+        // Map<String, String> options = context.getCatalogTable().getOptions();
```
Comment (EMsnap) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hbase/src/main/java/org/apache/inlong/sort/hbase/HBase2DynamicTableFactory.java:80]: Please remove useless code

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/utils/HBaseContainer.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+/** Standalone containerized HBase instance that builds the image on the fly. */
+@SuppressWarnings("rawtypes")
+public class HBaseContainer extends GenericContainer<HBaseContainer> {
```
Comment (EMsnap) [inline inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/utils/HBaseContainer.java:30]: Please add the container when implementing the IT, now it is not used

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/resources/flinkSql/hbase_e2e.sql [hunk 5]
```diff
@@ -0,0 +1,33 @@
+CREATE TABLE MyHBaseSource (
```
Comment (EMsnap) [inline inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/resources/flinkSql/hbase_e2e.sql:1]: ditto

**Review by EMsnap — APPROVED** [review]
LGTM

**Review by vernedeng — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by yunqingmoswu — APPROVED** [review]

**Pull request merged**